common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -1228,6 +1228,47 @@ static jlong NativeCrypto_EVP_parse_private_key(JNIEnv* env, jclass, jbyteArray
     return reinterpret_cast<uintptr_t>(pkey.release());
 }
 
+
+static jbyteArray NativeCrypto_EVP_raw_X25519_private_key(
+        JNIEnv* env, jclass cls, jbyteArray keyJavaBytes) {
+    CHECK_ERROR_QUEUE_ON_RETURN;
+    JNI_TRACE("NativeCrypto_EVP_raw_X25519_private_key(%p)", keyJavaBytes);
+
+    jlong key_ptr = NativeCrypto_EVP_parse_private_key(env, cls, keyJavaBytes);
```

Comment:
Looks like this function ignores trailing data. Comment above says it's intentional, but pointing out here in case carrying it over to this function was unintentional.

(In hindsight, we probably should have made our functions check for trailing data. If you want to allow trailing data, you can always do extra work, and forgetting to check is common. Ah well.)

Reply:
Yeah, the original unit tests caught that and I read the comment and adjusted the public key parsing behaviour to match.

Not because I think it's good, but because if other JCA implementations do this then I guess we don't really want to throw on encodings that they accept.